### DISCUSSION TITLE
Log config geht nicht

### DISCUSSION THREAD

**Discussion by Easton97-Jens** [ZoeyVid/NPMplus#552] [Q&A] [answered]
moin 
ich habe nach der anleiteung unter den discussions die config gesetzt in dem fad /data/ngnix/log.conf die wird bei einem neustart gelert so das kein inhalt mehr drin ist. Wenn ich sie neu erstelle kann ich sie in docker mir anschauen und sehe auch den gleichen inhalt nur nich nach einem neustart.

Wie kann ich trotdem den access_log setzen sodas ich das für auswertung für crowdsec  und andere sachen verwenden kann
sonst bin ich mit disem nginx proxy manager zufrinder wie mit dem anderen den es noch gibt

**Comment by Zoey2936** [comment]
nutzt du die neust version?

**Reply by Easton97-Jens** [reply]
ja
[docker-compose.txt](https://github.com/ZoeyVid/NPMplus/files/13798696/docker-compose.txt)
so sieht meine docker yml aus

**Reply by Zoey2936** [reply]
wenn ich:

```
access_log /data/nginx/access.log;
```
bei mir nach /opt/npm/nginx/custom/server_redirect.conf packe dann bleibt der log auch nach dem neustart

**Reply by Easton97-Jens** [reply]
ok legt er die log selber an oder muss ich sie ihm geben

**Reply by Zoey2936** [reply]
Wie? Ich füge den Inhalt in die Datei ein, und führe `docker compose up -d --force-recreate --pull always` aus und nach ein paar sekunden ist der log da

**Reply by Easton97-Jens** [reply]
er möche ihrgen wie patu nicht die log anlegen oder das gleiches ich habs auch einmal kommplett neu angelegt und selbst da macht er es nicht. kann ich das ihrgen wo sehen das er das geändert hatt da ich in der ngnix config seht es immer auf off wenn ich mit portainer mir das anschaue.

**Reply by Zoey2936** [reply]
die nginx.conf wird ignoriert, wenn du das in die custom conf packst, die customs confs wurden wegenem nem bug in alten versionen immer geleert, in der aktuellen ist das nimmer so und in meinen tests geht alles

**Reply by Easton97-Jens** [reply]
der eintrag pleit besteh blos er legt die datai /data/nginx/access.log nicht an ist egel wie lange ich warte

**Reply by Zoey2936** [reply]
kommen den anfragen die es logen kann?

**Reply by Easton97-Jens** [reply]
wie kann ich das überprüfen.
wenn ich das mit Wireshark über wache kommen anfragen an und werden auch bearbeitet von dem npm
ich kenne blos nicht den unterschied von einem host und bridg oder eigenes netwerk
ich komme von innen und auser halbe meine netwerk drauf zu so das er das eigenlich merken müste
kann ich in dem gleichen auch das log formart setzen

**Reply by Zoey2936** [reply]
Kannst du auch machen, in welcher datei setzt du den accsses log genau?

**Reply by Easton97-Jens** [reply]
in der die oben beschrieben hast

**Reply by Easton97-Jens** [reply]
[server_redirect.txt](https://github.com/ZoeyVid/NPMplus/files/13798928/server_redirect.txt)
so siht die aktuelle aus bei mir

**Reply by Zoey2936** [reply]
Greifst du auch auf einen redurect host zu?

**Reply by Easton97-Jens** [reply]
das ganze lauft in proxmox auf einem container nicht einer vm

**Reply by Zoey2936** [reply]
In den access log wird nur was reingeschrieben wenn du auf ein redirect host zugreifst aktuell

**Reply by Easton97-Jens** [reply]
wenn ich das auch in die anderen reinschreb kann ich alles log oder gibt es da eine algemeine config für

**Reply by Zoey2936** [reply]
muss in alle server_*.conf datein

**Reply by Easton97-Jens** [reply]
danke hatt jetz geklept

**Reply by Easton97-Jens** [reply]
Geht das auch mit einem Stream da ich fehler bekomme das die Konfiguration fehler haft ist

**Reply by Zoey2936** [reply]
müsste in die server_stream.conf
https://nginx.org/en/docs/stream/ngx_stream_log_module.html

**Reply by Easton97-Jens** [reply]
da habe ich ein Problem das ich kein log Formart setzen kann unknown log format "proxy" in /data/nginx/custom/server_stream.conf:1
das kommt als fehler und ich habe in  der stream.conf das log Formart gesetzt und ich kann das nicht direckt in der server_stream.conf weil er dan mit einem zeichen Probleme hatt

**Reply by Zoey2936** [reply]
Dann hast du was falsch konfiguriert in den custom confs

**Reply by Easton97-Jens** [reply]
dann frage ich mich wo ich das log formart setzen muss
log format is not specified in /data/nginx/custom/server_stream.conf:5 das kommt wenn ich nichts angebe

**Reply by Zoey2936** [reply]
![grafik](https://github.com/ZoeyVid/NPMplus/assets/75573284/36c15670-ddc7-4490-8f1a-25c8957b7976)
![grafik](https://github.com/ZoeyVid/NPMplus/assets/75573284/9bbb161e-aa61-4367-9271-45901d4d965a)
machs mal in die stream.conf statt in die server_stream.conf

**Reply by Easton97-Jens** [reply]
ich hatte das auch schon alle in der stream.conf da hatt er ein verbindung die ich ausbrobiert habe nicht protokoliert. kann das sein das die lade reinfolge in der nginx.conf falsch herum ist?
ich bin nur auf die idee gekommen da ich in der http_top,conf mein log formart für die server_dead, server_proxy und server_redirect  definierthabe und das hatt geklapt

**Reply by Zoey2936** [reply]
Mach mal `docker exec npmplus nginx -T`

**Reply by Zoey2936** [reply]
Und schau ob deine confs drinne sind

**Reply by Easton97-Jens** [reply]
die konfig lädt er rein die config die in stream  liegen intressiert das ganze wenig was in der stream.conf steht da die config da nicht reingeladen.
Er erstell die log sofort wenn das ganze in der stream.conf steht schreibt nichts rein was mit den configs ind dem Stream ordner ist

**Reply by Zoey2936** [reply]
kann sein, warte

**Reply by Zoey2936** [reply]
I füge eine stream_top.conf hinzu

**Reply by Zoey2936** [reply, marked as answer]
versuchs mal mit der neuen version und der stream_top.conf

**Reply by Easton97-Jens** [reply]
funktioniert vielen dank

**Reply by Easton97-Jens** [reply]
ngx_stream_log_module ist dises modul aktiv da ich so kein Fehler mehr bekomme er nichts protokoliert

**Reply by Zoey2936** [reply]
Wäre das modul nicht aktiv würde nginx dir sage, unknoen directive

**Reply by Easton97-Jens** [reply]
Danke für die Hilfe das funktioniert so gut das ich mich heute sogar ausgesperrt habe
